src/System.Numerics.Vectors/tests/Performance/Vector2/Normalize.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+
+namespace System.Numerics.Tests
+{
+    public static partial class Perf_Vector2
+    {
+        [Benchmark(InnerIterationCount = VectorTests.DefaultInnerIterationsCount)]
+        public static void NormalizeBenchmark()
+        {
+            var expectedResult = new Vector2(-16777216.0f, 16777216.0f);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                Vector2 result;
+
+                using (iteration.StartMeasurement())
+                {
+                    result = NormalizeTest();
+                }
+
+                VectorTests.AssertEqual(expectedResult, result);
+            }
+        }
+
+        public static Vector2 NormalizeTest()
+        {
+            var result = VectorTests.Vector2Value;
+
+            for (var iteration = 0; iteration < Benchmark.InnerIterationCount; iteration++)
+            {
+                result += Vector2.Normalize(result);
```

Comment:
I don't think its particularly safe to drop changing both the input and the output. That is what caused some of the intrinsics to be dropped already.

And while it isn't the case today that the call to `Normalize()` is optimized away, that just means that if the JIT is further optimized, it will get dropped and will look like a perf win.